py4web/utils/grid.py

```diff
@@ -780,14 +788,33 @@ def render_table_header(self):
                 #  add the sort order query parm
                 sort_query_parms = dict(self.query_parms)
 
+                attr = {}
+                if self.param.htmx_target:
+                    attr["_hx-target"] = self.param.htmx_target
+                    attr["_hx-swap"] = "innerHTML"
+
                 if key == sort_order:
                     sort_query_parms["orderby"] = "~" + key
-                    href = URL(self.endpoint, vars=sort_query_parms)
-                    col = A(heading, up, _href=href)
+                    url = URL(self.endpoint, vars=sort_query_parms)
+                    if self.param.htmx_target:
+                        rf = parse_referer(request)
+                        if rf and rf.query:
+                            url += "?%s" % rf.query
+                        attr["_hx-get"] = url
+                    else:
+                        attr["_href"] = url
+                    col = A(heading, up, **attr)
                 else:
                     sort_query_parms["orderby"] = key
-                    href = URL(self.endpoint, vars=sort_query_parms)
-                    col = A(heading, dw if "~" + key == sort_order else "", _href=href)
+                    url = URL(self.endpoint, vars=sort_query_parms)
+                    if self.param.htmx_target:
+                        rf = parse_referer(request)
+                        if rf and rf.query:
+                            url += "?%s" % rf.query
+                        attr["_hx-get"] = url
```

Comment:
Unsure about making py4web dependent on htmx. Could we maybe come up with a grid plugin system to configure this?